homeassistant/components/nuheat/__init__.py

```diff
@@ -95,7 +105,22 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
         _LOGGER.error("Failed to login to nuheat: %s", ex)
         return False
 
-    hass.data[DOMAIN][entry.entry_id] = (api, serial_number)
+    async def _async_update_data():
+        """Fetch data from API endpoint."""
+        try:
+            await hass.async_add_executor_job(thermostat.get_data)
+        except Exception as err:
```

Comment:
DataUpdateCoordinator technically would catch the generic exception here. I guess you get a bit mlre control of message this way though?

https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/update_coordinator.py#L172

Reply:
If I don't trap it, its going to fill the log with exception when nuheat does maintenance (which they do too often)

Reply:
fair point

Reply:
If there's an expected exception we should catch that expected exception. `Exception` is only for unexpected exceptions.

Reply:
I'll track down the list of exceptions it can throw since it looks like its always been `Exception` (before I moved it).

Reply:
https://github.com/home-assistant/core/pull/42127